### DISCUSSION TITLE
Use Android BLE Transmitter ID attribute to track Android phone and watch

### DISCUSSION THREAD

**Discussion by moblsu** [ESPresense/ESPresense#880] [Q&A]
I did some searching, but might not have used the correct parameters to find an active string. Sorry, if I missed it.

Since the Android phone seems to have a dynamic ID, has anyone used the "ID" attribute from the "BLE Transmitter" (active through the companion app) on Android to integrate into the YAML configuration for the "device_id" and "state_topic"?

In templating, I can get the ID from my device:
{{ state_attr('sensor.wXXXXv_ble_transmitter','id')}} = 1dXXXX16-481e-4579-8b35-ffcXXXXXX758_100_1

I can also pass that through templating to get the distance from my phone (I have BLE monitoring active on the phone too):
{{ state_attr('sensor.sXXXXu_beacon_monitor',state_attr('sensor.wXXXXv_ble_transmitter','id'))}}
{{ state_attr('sensor.sXXXXu_beacon_monitor','1dXXXX16-481e-4579-8b35-ffcXXXXXX758_100_1')}}

Can I do something similar in the YAML? I don't know enough about templating, when you can use it, or how to concatenate strings, etc.

- platform: mqtt_room
  device_id:  {{state_attr('sensor.sXXXXu_beacon_monitor',state_attr('sensor.wXXXXv_ble_transmitter','id'))}}
  name: 'bleWatch'
  state_topic: 'espresense/devices/' {{state_attr('sensor.sXXXXu_beacon_monitor',state_attr('sensor.wXXXXv_ble_transmitter','id'))}}
  timeout: 60
  away_timeout: 120

Even if I make a sensor with the ID as the state, I'm not sure how to incorporate it into the YAML for the mqtt_room platform.

edit:
I tried it, and found one flaw. The ID in MQTT Explorer ends in 758-100-1 not 758_100_1... So, I'd either have to truncate the string or replace the "_" with "-". I'm not sure how to do that either.

**Comment by Terastar-Paperclip** [comment]
This one's listed on our enrollment reference (#2324) as the "generic Android phone — hard" row, with this thread as the active spot for it. Android doesn't give the same static-identifier path Apple's IRK does, so there's no clean recipe today — if you get the BLE Transmitter ID approach working, please post the recipe here, we'd like to fold it into the docs.
